src/archive/src/main.rs

```diff
@@ -680,8 +685,29 @@ async fn status() -> ArchiveStatus {
 /// tools, can seamlessly integrate with it. The concrete interface (method name etc.) is
 /// provisional, but works.
 #[query]
+#[candid_method(query)]
 fn __get_candid_interface_tmp_hack() -> String {
     include_str!("../archive.did").to_string()
 }
 
 fn main() {}
+
+// Order dependent: do not move above any function annotated with #[candid_method]!
```

Comment:
If the order is moved the `archive.did` changes and needs to be updated, or does it break more things?

Reply:
If the order is changed, the test will fail because the canister interface generated from the `.did` will be incomplete.

Reply:
Ah, why incomplete? Won't it simply be reordered?

Reply:
No, there is some weird state handling going on in the rust macros. There is some global state that both the `candid::candid_method` annotations and the `candid::export_service!` access. And since macros are executed in order of appearance, anything after `candid::export_service!` will not be included in the generated `.did`. This then means that it is not a proper subtype (since it's missing methods) and causes the test to fail.